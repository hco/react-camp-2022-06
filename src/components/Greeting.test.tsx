import { render, screen } from "@testing-library/react";
import Greeting from "./Greeting";

describe("<Greeting />", () => {
  test("it renders without crashing", () => {
    render(<Greeting name="Kollegen von Stefan" />);
  });

  test("the given person will be greeted", () => {
    render(<Greeting name="Sarah" />);

    screen.getByText("Hello, Sarah!");
  });

  test("the name of the person is printed", () => {
    render(<Greeting name="Sarah" />);

    // screen.getByText("sarah", { exact: false });
    screen.getByText(/sarah/i);
  });

  test("renders a greeting", () => {
    render(<Greeting name="Sarah" />);

    screen.getByText(/hello/i);
  });

  test("it renders a paragraph with a greeting inside", () => {
    const { container } = render(<Greeting name="Philip" />);

    // Das hier entspricht nicht den von der testing-library vorgeschlagenen best practices.
    // testing-library rät dazu, keine implementierungsdetails zu testen,
    // sondern eine Komponente so zu testen, wie der user das wünscht.

    // eslint-disable-next-line testing-library/no-node-access
    expect(container?.firstChild?.nodeName).toBe("P");
    // eslint-disable-next-line testing-library/no-node-access
    expect(container?.firstChild).toHaveTextContent("Hello, Philip!");

    // screen.debug(container);
  });
});
