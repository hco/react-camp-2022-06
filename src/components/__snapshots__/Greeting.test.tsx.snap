// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting /> it renders according to snapshot 1`] = `
<div>
  <p>
    Hello, 
    Darko
    !
  </p>
</div>
`;
